### PR TITLE
MAGE-926: JS folders reorg

### PR DESCRIPTION
**Pull request by damcou**
This PR applies the changes suggested here https://github.com/algolia/algoliasearch-magento-2/pull/1506 by @sgeleon (with an extra fix for a logo path) targetting the 3.14.0-beta branch.

### CODE REVIEW

File: view/frontend/layout/algolia_search_handle.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0"?>
 <page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
     <head>
-        <css src="Algolia_AlgoliaSearch::internals/autocomplete.css" />
-        <css src="Algolia_AlgoliaSearch::internals/grid.css" />
-        <css src="Algolia_AlgoliaSearch::internals/algolia-reset.css" />
-        <css src="Algolia_AlgoliaSearch::internals/instantsearch.v3.css" />
-        <css src="Algolia_AlgoliaSearch::internals/recommend.css" />
+        <css src="Algolia_AlgoliaSearch::js/internals/autocomplete.css" />
+        <css src="Algolia_AlgoliaSearch::js/internals/grid.css" />
```
Comment (cammonro) [inline view/frontend/layout/algolia_search_handle.xml:5]: @damcou If we are aiming to follow [Magento's guidelines](https://developer.adobe.com/commerce/php/development/build/component-file-structure/#common-directories-1) it seems to me we should move CSS to a `css` directory. 

This is typically used for housing Less files in the `frontend` but I see straight CSS used in other places like in the `adminhtml` area in `Magento_GroupedProduct` and `Magento_TwoFactorAuth`. If we decide to switch to Less that would also be a more practical placement.

File: view/frontend/requirejs-config.js [hunk 2]
```diff
@@ -2,29 +2,29 @@ var config = {
...
-            'algoliaInstantSearch': 'Algolia_AlgoliaSearch/instantsearch',
-            'algoliaInsights'     : 'Algolia_AlgoliaSearch/insights',
-            'algoliaHooks'        : 'Algolia_AlgoliaSearch/hooks',
+            'algoliaCommon'       : 'Algolia_AlgoliaSearch/js/internals/common',
+            'algoliaAutocomplete' : 'Algolia_AlgoliaSearch/js/autocomplete',
+            'algoliaInstantSearch': 'Algolia_AlgoliaSearch/js/instantsearch',
+            'algoliaInsights'     : 'Algolia_AlgoliaSearch/js/insights',
+            'algoliaHooks'        : 'Algolia_AlgoliaSearch/js/hooks',
 
             // Autocomplete templates
```
Comment (cammonro) [inline view/frontend/requirejs-config.js:11]: @damcou If we have to move `template` anyway (which will disrupt downstream customization of Autocomplete and Recommend) then it makes sense to me to move this outside of `internals` as well. These file will be regularly customized and referenced so it never really made sense to me to include them under `internals`. 

We also already have a `template` directory (currently for an unused backend render implementation I think). Having two `template` directories might be a little confusing for some. What do you think?

**Review by cammonro — CHANGES_REQUESTED** [review]

File: view/frontend/requirejs-config.js (hunk 2 above)
Comment (damcou) [reply view/frontend/requirejs-config.js:11]: This makes total sense your right, adding the change 👌

**Comment by damcou** [comment]
Thanks for the feedbacks @cammonro , you're right, the folder structure looks better this way :)

**Review by cammonro — APPROVED** [review]
Awesome! Looks good.

**Pull request merged**